packages/teuchos/numerics/src/Teuchos_SerialSymDenseMatrix.hpp

```diff
@@ -1185,6 +1196,35 @@ std::ostream& operator<< (std::ostream& os, const Teuchos::SerialSymDenseMatrix<
 }
 #endif
 
+/// \brief Ostream manipulator for SerialSymDenseMatrix 
+template<typename OrdinalType, typename ScalarType>
+struct SerialSymDenseMatrixPrinter {
+public:
+  const SerialSymDenseMatrix<OrdinalType,ScalarType> &obj;
+  SerialSymDenseMatrixPrinter(
+        const SerialSymDenseMatrix<OrdinalType,ScalarType> &obj_in)
+      : obj(obj_in) {}
+};
+
+/// \brief Output SerialSymDenseMatrix object through its stream manipulator. 
+template<typename OrdinalType, typename ScalarType>
+std::ostream&
+operator<<(std::ostream &out,
+           const SerialSymDenseMatrixPrinter<OrdinalType,ScalarType> printer)
+{
+  printer.obj.print(out);
+  return out;
+}
+
+/// \brief Return SerialSymDenseMatrix ostream manipulator Use as:
+template<typename OrdinalType, typename ScalarType>
+SerialSymDenseMatrixPrinter<OrdinalType,ScalarType>
+Print(const SerialSymDenseMatrix<OrdinalType,ScalarType> &obj)
```

Comment:
Since this is a function and not a type, can we call it `print()` instead of `Print()`?  The convention used in Teuchos is:

* Functions: `startWithLowerCase()`
* Classes/Types: `StartWithUpperCase()`

Reply:
There is a struct in Teuchos namespace called print (In Teuchos_any.hpp); it doesn't follow the convention, either.  The compiler complains about having different types of thing called print.  I guess you can't overload a struct with a function.  So I had to give this function a different name.

Reply:
> There is a struct in Teuchos namespace called print (In Teuchos_any.hpp)

Shoot. That stuff got copied many years ago and we (me) did not rename the types to match Teuchos convention.  That is on me.

 How about  `printMat()` then?  That makes it clear this is not some type of built-in print() function.  Otherwise, we can just go with 'Print()' and deal with tis.

Reply:
I can do printMat.  Stay tuned.

Reply:
Done.

Reply:
Thanks!